### PR TITLE
feat(workflow): add Sync CSI Repos with Upstream action

### PR DESCRIPTION
**Pull request by derekbit**

### CODE REVIEW

File: scripts/sync-csi-repos-with-upstream.sh [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    git clone ${TARGET_REPO_PATH}
+    pushd ${target_repo}
+
+    # Sync branches
+    git remote add src_remote ${SOURCE_REPO_PATH}
+    git fetch src_remote --quiet
+    git push origin "refs/remotes/src_remote/*:refs/heads/*" --force
+
+    # Sync tags
+    git tag -d $(git tag -l)
```
Comment (innobead) [inline scripts/sync-csi-repos-with-upstream.sh:30]: Will this delete our downstream tags? Do we need to delete all tags first?

**Review by c3y1huang — APPROVED** [review]
LGTM

File: scripts/sync-csi-repos-with-upstream.sh (hunk 1 above)
Comment (innobead) [reply scripts/sync-csi-repos-with-upstream.sh]: Just make sure the downstream tags need to be kept after syncing from the source repo.

Comment (derekbit) [reply scripts/sync-csi-repos-with-upstream.sh]: The script won't delete any tags in the target repo (our forked repo).

**Review by innobead — APPROVED** [review]

**Pull request merged**